src/entities/join/parser.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseJoin } from '.';
+import { WeightedJoin } from './weighted/weightedJoin';
+
+/*********************** Basic Helper to get join class from pool type *************/
+export type JoinConfig = {
+    customPoolJoins: Record<string, BaseJoin>;
+};
+
+export class JoinParser {
+    private readonly poolJoins: Record<string, BaseJoin> = {};
+
+    constructor(config?: JoinConfig) {
+        const { customPoolJoins } = config || {};
+        this.poolJoins = {
+            Weighted: new WeightedJoin(),
```

Comment:
This pattern needs revisiting, although doesn't need to be done in this PR. How do I as a consumer of the SDK know that the weighted pool is keyed on `Weighted`?